src/topNavigation/topNavigation.tsx

```diff
@@ -31,139 +38,153 @@ interface State {
   favouritesCount: number;
 }
 
-/*
-// TODO: Not currently possible due to the way HDS handles the active class name, but possibly in the future.
-// For now, the same structure is defined separately for every link further below instead.
-
 interface TopNavigationLinkProps {
-  to: string,
-  label: ReactNode,
-  className?: string
+  to: string;
+  label: string;
+  className?: string;
 }
 
-const TopNavigationLink = ({to, label, className}: TopNavigationLinkProps): JSX.Element => {
+const TopNavigationLink = ({
+  to,
+  label,
+  className,
+}: TopNavigationLinkProps): JSX.Element => {
   const match = useMatch(to);
 
-  return <Navigation.Item
+  return (
+    <Header.Link
       label={label}
       as={Link}
       to={to}
       active={match !== null}
       className={className}
-  />;
+    />
+  );
 };
- */
 
 const TopNavigation = ({
   openLoginModal,
   favouritesCount,
   user,
 }: TopNavigationProps): JSX.Element => {
-  const returnHome = (navigate: NavigateFunction) => {
-    navigate(getRouteById(AppRoutes.HOME));
-  };
-
   const { t, i18n } = useTranslation();
-
   const navigate = useNavigate();
 
-  const matchPlotSearchAndCompetitions = useMatch(
-    getRouteById(AppRoutes.PLOT_SEARCH_AND_COMPETITIONS) + '/*'
-  );
-  const matchOtherCompetitionsAndSearches = useMatch(
-    getRouteById(AppRoutes.OTHER_COMPETITIONS_AND_SEARCHES) + '/*'
-  );
-  const matchAreaSearch = useMatch(
-    getRouteById(AppRoutes.AREA_SEARCH_LANDING) + '/*'
-  );
-  const matchFavourites = useMatch(getRouteById(AppRoutes.FAVOURITES));
-  const changeLanguage = (language: Language) => {
-    i18n.changeLanguage(language).then(() => {
-      document.location.reload();
-    });
+  const naviLinks: TopNavigationLinkProps[] = [
+    {
+      to: getRouteById(AppRoutes.PLOT_SEARCH_AND_COMPETITIONS),
+      label: t(
+        'topNavigation.tabs.plotSearchAndCompetitions',
+        'Plot search and competitions',
+      ),
+    },
+    {
+      to: getRouteById(AppRoutes.OTHER_COMPETITIONS_AND_SEARCHES),
+      label: t(
+        'topNavigation.tabs.otherCompetitionsAndSearches',
+        'Other competitions and searches',
+      ),
+    },
+    {
+      to: getRouteById(AppRoutes.AREA_SEARCH_LANDING),
+      label: t('topNavigation.tabs.areaSearch', 'Area search'),
+    },
+  ];
+
+  const languages: LanguageOption[] = [
+    {
+      label: t('language.fi', 'Finnish'),
+      value: Language.FI,
+      isPrimary: false,
+    },
+    {
+      label: t('language.se', 'Swedish'),
+      value: Language.SV,
+      isPrimary: false,
+    },
+    {
+      label: t('language.en', 'English'),
+      value: Language.EN,
+      isPrimary: false,
```

Comment:
I'd say that if we want to always show the names of the languages in the language menu in that language itself, which is common and good practice, these are a rare exception where they outright shouldn't be translatable.